Development/nmos-cpp-node/node_implementation.cpp

```diff
@@ -894,6 +920,374 @@ void node_implementation_init(nmos::node_model& model, slog::base_gate& gate)
         auto channelmapping_output = nmos::make_channelmapping_output(id, name, description, source_id, channel_labels, routable_inputs);
         if (!insert_resource_after(delay_millis, model.channelmapping_resources, std::move(channelmapping_output), gate)) throw node_implementation_init_exception();
     }
+
+    // examples of using IS-12 control protocol
+    // they are based on the NC-DEVICE-MOCK
+    // See https://specs.amwa.tv/nmos-device-control-mock/#about-nc-device-mock
+    // See https://github.com/AMWA-TV/nmos-device-control-mock/blob/main/code/src/NCModel/Features.ts
+    if (0 <= nmos::fields::control_protocol_ws_port(model.settings))
+    {
+        // example to create a non-standard Gain control class
+        const auto gain_control_class_id = nmos::make_nc_class_id(nmos::nc_worker_class_id, 0, { 1 });
+        const web::json::field_as_number gain_value{ U("gainValue") };
+        {
+            // Gain control class properties
+            std::vector<web::json::value> gain_control_properties = { nmos::experimental::make_control_class_property(U("Gain value"), { 3, 1 }, gain_value, U("NcFloat32")) };
+
+            // create Gain control class
+            auto gain_control_class = nmos::experimental::make_control_class(U("Gain control class descriptor"), gain_control_class_id, U("GainControl"), gain_control_properties);
```

Comment:
I'm getting confused between classes that create the class definitions and those that create the class instances.

In line 938 we have a reference to `make_control_class` that is creating a class descriptor (i.e. class definition) and in line 948 we have `make_nc_worker` that is creating a class instance of worker. 

Perhaps to avoid confusion the `make_control_class` should be `make_control_class_desc`?  This would relate back to the definition of functions in `control_protocol_state`

Reply:
done